### PR TITLE
feat(agw): Scaffold S1AP state manager proto service.

### PR DESCRIPTION
**Pull request by mstre123**
Signed-off-by: mstre123 <mstre123@gmail.com>

## Summary
Implement and generate protos for s1ap state manager grpc service.
#8834

### CODE REVIEW

File: lte/protos/s1ap_state_manager.proto [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2020 The Magma Authors.
```
Comment (markjen) [inline lte/protos/s1ap_state_manager.proto]: nit: 2021 :)

File: lte/protos/s1ap_state_manager.proto [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import "lte/protos/oai/s1ap_state.proto";
+
+package magma.lte;
+
+option go_package = "magma/lte/cloud/go/protos";
+
+service S1apStateManagerService {
+  // Connect handles an S1AP Connection Request
+  rpc Connect (ConnectRequest) returns (ConnectResponse) {}
+  // Disconnect handles an S1AP DisconnectRequest
```
Comment (mstre123) [inline lte/protos/s1ap_state_manager.proto]: typo

File: lte/protos/s1ap_state_manager.proto [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  rpc Disconnect (DisconnectRequest) returns (DisconnectResponse) {}
+}
+
+message ConnectRequest {
+  oai.S1apState state = 1;
+}
+
+message ConnectResponse{
+  uint32 associd = 1; // s1ap association id
```
Comment (themarwhal) [inline lte/protos/s1ap_state_manager.proto]: this is more just my preference but can we place the comments above the field? That's what the uber style guide for proto seems to use: https://docs.magmacore.org/docs/next/contributing/contribute_conventions#grpc :) 
```suggestion
  // s1ap association id
  uint32 associd = 1;
```

Comment (mstre123) [reply lte/protos/s1ap_state_manager.proto]: Haha I actually prefer them that way!
I was following the pattern in the proto i was importing https://github.com/magma/magma/blob/6e862525c76c143ea07dbf18d1e1c23655c4d64b/lte/protos/oai/s1ap_state.proto#L58
but will move the comments up.

Comment (themarwhal) [reply lte/protos/s1ap_state_manager.proto]: yeah I think we're definitely not consistent as ususal :D hopefully we can push to converge 😄

File: lte/protos/s1ap_state_manager.proto [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+service S1apStateManagerService {
+  // Connect handles an S1AP Connection Request
+  rpc Connect (ConnectRequest) returns (ConnectResponse) {}
+  // Disconnect handles an S1AP DisconnectRequest
+  rpc Disconnect (DisconnectRequest) returns (DisconnectResponse) {}
+}
+
+message ConnectRequest {
+  oai.S1apState state = 1;
```
Comment (markjen) [inline lte/protos/s1ap_state_manager.proto]: i don't think we'll want to send this entire thing to setup a single enb assoc

File: lte/protos/s1ap_state_manager.proto (hunk 3 above)
Comment (markjen) [inline lte/protos/s1ap_state_manager.proto]: i think assoc_id is usually snake_cased

File: lte/protos/s1ap_state_manager.proto [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+limitations under the License.
+*/
+
+syntax = "proto3";
+
+import "lte/protos/oai/s1ap_state.proto";
+
+package magma.lte;
+
+option go_package = "magma/lte/cloud/go/protos";
```
Comment (markjen) [inline lte/protos/s1ap_state_manager.proto]: do we need this option? can we go without?

**Review by markjen — CHANGES_REQUESTED** [review]
consider putting this in /src/protos since it's a new service and unlikely to be used by other stuff in /lte/gateway

File: lte/protos/s1ap_state_manager.proto (hunk 1 above)
Comment (mstre123) [reply lte/protos/s1ap_state_manager.proto]: good catch - been copy and pasting that from other files - will set auto heading in my ide

File: lte/protos/s1ap_state_manager.proto (hunk 3 above)
Comment (mstre123) [reply lte/protos/s1ap_state_manager.proto]: yeah i copied that from from S1apState
https://github.com/magma/magma/blob/6e862525c76c143ea07dbf18d1e1c23655c4d64b/lte/protos/oai/s1ap_state.proto#L59
but ill fix move it to camel case

File: lte/protos/s1ap_state_manager.proto (hunk 5 above)
Comment (mstre123) [reply lte/protos/s1ap_state_manager.proto]: was following the pattern we've used in other proto files.
i think the style guide prefers this to just defining it in the protoc call
https://developers.google.com/protocol-buffers/docs/reference/go-generated#package

File: lte/protos/s1ap_state_manager.proto (hunk 4 above)
Comment (mstre123) [reply lte/protos/s1ap_state_manager.proto]: changed to just store the id

**Pull request closed without merging**